zap/src/main/java/org/zaproxy/zap/utils/ZapNumberSpinner.java

```diff
@@ -55,6 +59,19 @@ public ZapNumberSpinner(int minValue, int defaultValue, int maxValue) {
                 .getTextField()
                 .setFormatterFactory(
                         new DefaultFormatterFactory(new ZapNumberFormatter(minValue, maxValue)));
+        DefaultEditor editor = (DefaultEditor) getEditor();
+        editor.getTextField()
+                .addFocusListener(
+                        new FocusAdapter() {
+                            @Override
+                            public void focusGained(final FocusEvent e) {
+                                SwingUtilities.invokeLater(
+                                        () -> {
+                                            JTextField tf = (JTextField) e.getSource();
+                                            tf.selectAll();
```

Comment:
This could use the new variable (which should make the lambda one statement).

Reply:
Okay I switched it to:
```java
                                SwingUtilities.invokeLater(
                                        () -> {
                                            textField.selectAll();
                                        });
```
But then sonar wants me to use a method reference. If I switch it to `JTextField::selectAll` then eclipse says that params aren't acceptable for invoke later... 🤷

Reply:
Will just leave it as a lambda for now.

Reply:
`textField::selectAll`